### DISCUSSION TITLE
Max number of channels to optimize for

### DISCUSSION THREAD

**Discussion by isaacmalka** [google/meridian#386] [Q&A]
Hi, in both LightWeightMMM and Meridian we want to know what is the optimal number of channels the model can handle for both finding channels' contributions to KPI, but more importantly - for the optimizing process. I found that having more than 15 channels and given my current infra setup, it gets stuck in optimization so perhaps it's better not to have more than that amount-- but was wondering if there are any guidelines from experiments you've conducted to provide us with some insights. Thanks.

**Comment by alexwakim** [comment]
Regarding the maximum number of channels Meridian can handle, there isn't a strict limit. It depends on your specific data, model specifications (`ModelSpec`), priors, and compute resources.

As the number of channels increases, the inferential burden of estimating many channels simultaneously increases. This might manifest in high variance estimates or even lack of convergence. There is also the computation complexity involved, as you mention. 

Here are some considerations:

- Data Size and Dimensionality: As the number of channels increases, the model becomes more complex. One must have enough data to support this complexity. See [Amount of data needed](https://developers.google.com/meridian/docs/advanced-modeling/amount-data-needed).
- Model Complexity: The number of channels is just one aspect of model complexity. To reduce complexity, consider reducing the number of knots or removing any control variables that are merely [predictor variables](https://developers.google.com/meridian/docs/advanced-modeling/control-variables#selecting-control-variables).
- Data Sparsity: If some channels have limited data or spend, it can be difficult to estimate its effect. This can affect inference throughout the model, since parameters are estimated jointly. With more channels, it’s more likely you’ve included such a channel.
 
Here are some recommendations:

- Explore grouping similar channels, particularly channels with low spend. 
- Consider more informative priors. More informative priors add information to the model and provide regularization. This can help with the increased inferential burden of more channels.
- Consider excluding channels with low spend. This can induce some bias, but if the channel’s spend is small enough the induced bias will be negligible. 
- If geo granular data is available, consider running a geo model. The geo-granular data adds a lot of data, which helps support more channels.
